### PR TITLE
fix(geometry): bbox_to_mask3d intersects axis ranges for a full/overhanging-axis box

### LINKED ISSUE #4255: Bug: `bbox_to_mask3d` fills the whole volume when a box spans or overhangs a full axis
`bbox_to_mask3d` builds its mask as a union of three axis slabs and then recovers the intersection with `all` reductions. When a box covers an entire output axis, the union intermediate is `True` everywhere, every reduction comes back `True`, and the function fills the whole volume, losing the bounds on the other two axes. A box that overhangs the volume on one axis, which is the normal state of a box after a crop or a translation, triggers the same failure. `Boxes3D.to_mask` on the same box fills the intended region on both its loop and export paths.

Reproduced 2026-09-05 at `1eea5835`, PyTorch 2.14.0, CPU.

## Reproduction

```python
import torch
from kornia.geometry.bbox import bbox_to_mask3d
from kornia.geometry.boxes import Boxes3D

size = (4, 4, 5)  # depth, height, width
for label, xyzxyz in {
    "x spans 0..4 (full width)": [0., 1., 1., 4., 2., 2.],
    "y spans 0..3 (full height)": [1., 0., 1., 2., 3., 2.],
    "z spans 0..3 (full depth)": [1., 1., 0., 2., 2., 3.],
    "x overhangs -1..5": [-1., 1., 1., 5., 2., 2.],
    "interior 1..2": [1., 1., 1., 2., 2., 2.],
}.items():
    b = Boxes3D.from_tensor(torch.tensor([xyzxyz]), "xyzxyz_plus")
    print(label, int(bbox_to_mask3d(b.data, size).sum()), int(b.to_mask(*size).sum()))
```

## Observed

```text
x spans 0..4 (full width)  80 20
y spans 0..3 (full height) 80 16
z spans 0..3 (full depth)  80 16
x overhangs -1..5          80 20
interior 1..2              8 8
```

The volume has 80 voxels. Every full-axis case fills all of them; only the interior box matches the method.

## Expected

The intersection of the three slabs, as `Boxes3D.to_mask` computes it: 20, 16, 16, 20 and 8 voxels. The three-way `|` followed by `all` reductions should be a three-way `&` of the broadcast slabs, which is what the export path of `Boxes3D.to_mask` already does.

## Cause

In `kornia/geometry/bbox.py`, `bbox_to_mask3d` computes `m = z_slab | y_slab | x_slab` and then `cond1 = m.all(dim=3).all(dim=2)` and its two siblings. For a slab `X` that covers every index, `all_{z,y}(Z | Y | X) = X | Zfull | Yfull` is `True` for every `x`, and the same happens to the other two conditions, so the product is `True` on the whole grid. For interior boxes the `all` reductions recover the slabs exactly, which is why the existing interior-box tests pass.

## Classification

Broken output. A wrong mask for a valid, in-range box. Not a rounding or convention question: the inclusive-truncating policy (#4015) and the `float32` channel-axis output (#4250) are separate.

## Disposition

**Fix directly, not in the window.** The canonical result is unambiguous and `Boxes3D.to_mask`'s export path already implements it; the repair changes no output for any box that the current code handles correctly. The conventions batch 4c PR (#4253) documents the current behavior in the `bbox_to_mask3d` warning and pins it: `test_wart_bbox_to_mask3d_fills_the_whole_volume_when_a_box_spans_an_axis_<this issue>` records today's output, and a strict `xfail` on the intersection result turns into an XPASS when this is fixed.

Found by a review of #4253 during the bbox/boxes conventions audit (#4020); the audit changes no runtime behavior.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

### PR DESCRIPTION
**Pull request by tritsystem**
## What does this pull request do?

Fixes #4255: `bbox_to_mask3d` fills the whole volume when a box spans or overhangs a full axis, instead of the intersection.

The old implementation built the mask as a union of three axis slabs (`|`) and tried to recover the intersection with a three-way `all()` reduction/product. That recovery breaks the moment any one slab covers a whole output axis: the union goes all-true on that axis and the reduction can no longer see the other two slabs' bounds. A box that overhangs the volume on one axis — the normal state of a box after a crop or a translation, not an edge case — triggers the same failure.

Computes the intersection directly with `&` instead, which needs no recovery step: each broadcast slab already carries its own axis's bound at every position, so the elementwise intersection is exactly the answer. `Boxes3D.to_mask`'s export path already does this; `bbox_to_mask3d` now matches it exactly.

## Related issue or discussion

Closes #4255.

## What did you check?

- All 4 parametrized cases from the issue (full width/height/depth, overhang) now match `Boxes3D.to_mask` exactly (20, 16, 16, 20 voxels), not 80.
- Interior boxes and the docstring's own worked example are byte-identical before and after — this only changes behavior for boxes that cover or overhang a full axis.
- Updated the two tests #4253 added for this exact issue: flipped the `strict=True` `xfail` on `test_convention_bbox_to_mask3d_intersects_the_axis_ranges_for_a_full_axis_box_4255` to a normal passing test (it would otherwise fail the suite as an unexpected XPASS), and corrected the parametrized "wart pin" test's stale `== 80.0` assertions to `== intersection` now that the behavior it was pinning is fixed.
- Removed the now-resolved `#4255` paragraph from the docstring's `.. warning::` block; the other tracked warts in that block (#4015, #4250, #4013, #4248) are unrelated and untouched.

### CODE REVIEW

**Review by ducha-aiki — CHANGES_REQUESTED** [review]
Review at `c64ea775` — implementation verified; documentation follow-up requested.

The mask implementation is correct: an exhaustive sweep of 58,320 ordered integer cuboids over a (4,4,5) volume matched a direct intersection oracle at head with zero mismatches; the merge base mismatched 31,347 cases. Focused bbox_to_mask3d tests passed (14 CPU float32/float64 cases), and all current GitHub checks are green.

Two closeout items remain:

1. `kornia/geometry/boxes.py:1451-1453` still publicly warns that bbox_to_mask3d fills the whole volume for a full/overhanging axis. This is false after the patch: remove that sentence and its #4255 link. The required issue-number closeout search also finds #4255 in the new comment at `kornia/geometry/bbox.py:434`; keep the explanation, but retire the resolved-issue reference under kornia/. The test pins can retain their issue references.
2. The CHANGELOG entry cites issue #4255 but does not cite PR #4303. Please add the PR number.

No production-code change is requested; I recommend approval after these documentation edits.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-6-astra).

**Comment by ducha-aiki** [comment]
Delta re-review at `8cdd3549` — **APPROVED, merge-ready**. The delta is exactly the three asks: the `Boxes3D.to_mask` warning loses its now-false "fills the whole volume" sentence and keeps the unrelated #4015 one, the `kornia#4255` tag is gone from the production comment in `bbox.py` (the explanation stays), and the CHANGELOG entry cites `(#4255, #4303)`. `grep -rnE "#4255|issues/4255" kornia/` is empty at this head; the one `tests/` hit is your pin naming its issue, as it should. Code is unchanged from the reviewed `c64ea775`. CI 38 green with the last leg in progress at the time of writing; `MERGEABLE`.

Posted on behalf of @ducha-aiki by Claude (Claude Fable 5.1).

**Review by ducha-aiki — DISMISSED** [review]
